recipe/meta.yaml

```diff
@@ -67,95 +66,120 @@ outputs:
         - cmake -S . -B ./build -G=Ninja && cmake --build ./build -v
     about:
       home: https://developer.nvidia.com/cuda-toolkit
-      license_file: LICENSE
+      license_file:
+        - {{ PREFIX }}/targets/{{ ctk_targets[target_platform] }}/LICENSE
       license: LicenseRef-NVIDIA-End-User-License-Agreement
       license_url: https://docs.nvidia.com/cuda/eula/index.html
       summary: Compiler for CUDA applications.
       description: |
         Compiler for CUDA applications.
       doc_url: https://docs.nvidia.com/cuda/index.html
 
-  - name: cuda-nvcc_{{ target_platform }}
+# Work around conda-build bug https://github.com/conda/conda-build/issues/2815
+# We must define a number of placeholder package names that will be ignored by
+# Jinja logic. The number of placeholder packages is the total number
+# generated by "for" loops minus the number of explicit "- name" entries (so
+# exclude one iteration). Currently there are 6 packages (3 platforms,
+# non-impl/impl package for each) and 2 explicit "- name" entries in the loop,
+# so we need 4 placeholders.
+{% if False %}
+  - name: placeholder-1
+  - name: placeholder-2
+  - name: placeholder-3
+  - name: placeholder-4
+{% endif %}
+
+{% for cross_target_platform, cross_target_dir in ctk_targets.items() %}
+  - name: cuda-nvcc_{{ cross_target_platform }}
     build:
-      noarch: generic
-    run_exports:
-      strong:
-        - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
+      run_exports:
+        strong:
+          - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
     files:
       - etc/conda/activate.d/~cuda-nvcc_activate.sh      # [linux]
       - etc/conda/deactivate.d/~cuda-nvcc_deactivate.sh  # [linux]
     requirements:
       run:
         - {{ pin_subpackage("cuda-nvcc-impl_" + target_platform, exact=True) }}
+        {% if cross_target_platform != target_platform %}
+        - {{ pin_subpackage("cuda-nvcc-impl_" + cross_target_platform, exact=True) }}
+        {% endif %}
     test:
       commands:
         - test -f $PREFIX/etc/conda/activate.d/~cuda-nvcc_activate.sh      # [linux]
         - test -f $PREFIX/etc/conda/deactivate.d/~cuda-nvcc_deactivate.sh  # [linux]
     about:
       home: https://developer.nvidia.com/cuda-toolkit
-      license_file: LICENSE
+      license_file:
+        - {{ PREFIX }}/targets/{{ cross_target_dir }}/LICENSE
       license: LicenseRef-NVIDIA-End-User-License-Agreement
       license_url: https://docs.nvidia.com/cuda/eula/index.html
       summary: Compiler for CUDA applications.
       description: |
         Compiler for CUDA applications.
       doc_url: https://docs.nvidia.com/cuda/index.html
 
-  - name: cuda-nvcc-impl_{{ target_platform }}
+  - name: cuda-nvcc-impl_{{ cross_target_platform }}
```

Comment:
This is wrong. cuda nvcc is natively a cross compiler (like clang unlike gcc), so you only have one compiler that works for all platforms. You can't have both the implementation and the cross compiler in the same feedstock. You should keep `cuda-nvcc-impl_{{ target_platform }}` in this feedstock and have a new feedstock for `cuda-nvcc_{{ cross_target_platform }}`.

Also `cuda-nvcc-impl_{{ target_platform }}` should only have the headers and the libraries